cpp/include/cudf/fixed_point/detail/floating_conversion.hpp

```diff
@@ -29,49 +31,6 @@
 namespace CUDF_EXPORT numeric {
 namespace detail {
 
-/**
```

Comment:
Moved to `count_significant_bits.hpp`

Reply:
Perhaps this header can be moved into `cudf/utililties` and use `cudf::detail` namespace instead of `numeric`?

Reply:
I considered moving it to `cudf/utilities` but it didn't seem like a good fit, at least as is, since `cudf/utilities` contains files named `<module_name>.hpp` containing utilities for that particular module. e.g. bit.hpp contains utilities for `bitmasks`. I can perhaps move this header as `cudf/utilities/numeric.hpp` so to indicate that it contains utilities related to `numeric`?

Reply:
The `numeric` namespace is really just for managing fixed-point types and this utility is not necessarily fixed-point specific as implemented but it feels a bit fragile to use it outside of fixed-point functions. 
It just was strange seeing the bitmask some code calling a fixed-point/numeric utility function.

You are only using it for an uint64 type so why not just replace the call site with `(64 - __builtin_clzll(block_size))` ?
Seems like we going through a lot of trouble for one line of code.

Reply:
There is some compiler specific behavior with `#ifdef __CUDA_ARCH__` in the utility which we will need to replicate here. Also, this utility is being used at two other locations in libcudf (io/comp and io/parquet) as well so I think it's better to keep it (avoid duplication especially the `#ifdef __CUDA_ARCH__` part) and reuse everywhere but I am not fully sure where to put it if not here.

Reply:
Ok. I would say these 2 other places are fragile as well since it is reasonable to expect this utility to change to suit numeric/fixed-point behavior and not general utility usage.
We can sort this out in a different PR. I was not expecting to fix this here.
I'd prefer removing the `count_significant_bits.hpp` and changing the code back or replacing the call site with the one line of code.

Reply:
Removed and reverted!